src/components/Banner/Banner.tsx

```diff
@@ -0,0 +1,141 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { isNotUndefined } from 'ramda-adjunct';
+
+import { BannerProps } from './Banner.types';
+import { BannerVariants } from './Banner.enums';
+import { Icon } from '../Icon';
+import { Button } from '../Button';
+import { Inline, Padbox } from '../layout';
+import colors from '../../theme/colors';
+import { IconTypes, SSCIconNames } from '../../theme/icons/icons.enums';
+import { getColor, pxToRem } from '../../utils';
+
+const colorVariants = {
+  [BannerVariants.info]: colors.radiantBlueberry,
+  [BannerVariants.warn]: colors.pumpkin,
+  [BannerVariants.error]: colors.strawberry,
+};
+
+const iconVariants = {
+  [BannerVariants.info]: SSCIconNames.infoCircle,
+  [BannerVariants.warn]: SSCIconNames.exclamationTriangle,
+  [BannerVariants.error]: SSCIconNames.ban,
+};
+
+const bgVariants = {
+  [BannerVariants.info]: { r: 235, g: 244, b: 252, a: 0.5 },
+  [BannerVariants.warn]: { r: 240, g: 143, b: 0, a: 0.05 },
+  [BannerVariants.error]: { r: 241, g: 67, b: 28, a: 0.05 },
+};
+
+const StyledInline = styled(Inline)<Pick<BannerProps, 'variant'>>`
+  border: ${({ variant }) => `1px solid ${colorVariants[variant]}`};
+  background: ${({ variant }) =>
+    `rgba(${bgVariants[variant].r}, ${bgVariants[variant].g}, ${bgVariants[variant].b}, ${bgVariants[variant].a})`};
+`;
+
+const IconContainer = styled.div<Pick<BannerProps, 'variant'>>`
+  background-color: ${({ variant }) => colorVariants[variant]};
+  height: 100%;
+  display: flex;
+  align-items: center;
+`;
+
+const StyledIcon = styled(Icon)`
+  color: ${colors.graphite5H};
+`;
+
+const CloseButton = styled.button`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border: none;
+  flex-shrink: 0;
+  cursor: pointer;
+  color: ${getColor('graphite3B')};
+`;
+
+const TimesIcon = styled(Icon)`
+  font-size: ${pxToRem(16)};
+`;
+
+const Banner: React.FC<BannerProps> = ({
+  children,
+  variant = BannerVariants.info,
+  cta1,
```

Comment:
I would rather refactor this into an array (let's call it 'actions') since we might need one or three CTA buttons in the future. We could reuse the logic and types (https://github.com/securityscorecard/design-system/blob/alpha/src/components/Datatable/BatchModule/BatchActions/BatchActions.types.ts#L4) from the `BatchModule` in Datatable for this.

Reply:
I see that AC stating there should not be more than two CTAs but the point still stands. Let's make it a `actions` array and we can type it as a tuple and restrict the number of items in the array.

```ts
type MaxLengthArray = readonly [string?, string?];

const a: MaxLengthArray = ['a', 'b']; // <-- this is ok
const b: MaxLengthArray = ['b']; // <-- this is ok
const c: MaxLengthArray = []; // <-- this is ok
const d: MaxLengthArray = ['a', 'b', 'c']; // <-- this will throw: Source has 3 element(s) but target allows only 2.
```

Reply:
@ajkl2533 I understand it would be cleaner but AC includes requirement "Each banner should not have more than 2 CTAs" and it makes kind of sense. User can still pass another CTAs within `children` tho. I can at least rename it to `action1` or `firstAction`. WDYT?

Reply:
@PrincAm I get it but:
1) with my suggestion we still satisfy the "2 CTAs at most" requirement
2) from my POV whenever you have to index some property you probably doing something wrong
3) when you split it into two props, users could think that each CTA has a different slot (I mean they are rendered in different positions) so they will try to put only `cta2` and that would be kind of strange
4) with the `Action` type we could also support dropdowns 
5) sooner or later there will be a requirement to have 3 CTAs and it's much easier to change the type than change the implementation

Reply:
You're right, nice clarification. I've used tuple and updated component api.